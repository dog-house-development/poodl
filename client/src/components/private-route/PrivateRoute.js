import React from "react";
import { Route, Redirect } from "react-router-dom";
import { connect } from "react-redux";
import PropTypes from "prop-types";

const propTypes = {
  auth: PropTypes.object.isRequired
};

const PrivateRoute = ({ component: Component, auth, ...rest }) => (
  <Route
    {...rest}
    render={props =>
      auth.isAuthenticated === true ? (
        <Component {...props} />
      ) : (
        <Redirect to="/login" />
      )
    }
  />
);

const mapStateToProps = state => ({
  auth: state.auth
});

PrivateRoute.propTypes = propTypes;
export default connect(mapStateToProps)(PrivateRoute);
