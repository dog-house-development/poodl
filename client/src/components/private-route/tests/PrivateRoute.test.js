import React from "react";
import ReactDOM from "react-dom";
import { shallow, configure } from "enzyme";
import Adapter from "enzyme-adapter-react-16";
import _ from "lodash";

import { PrivateRoute } from "../PrivateRoute";

configure({ adapter: new Adapter() });

let props, wrapper, instance;
beforeEach(() => {
  props = {
    auth: {
      isAuthenticated: true,
      loading: false,
      user: {
        exp: 1580940830,
        iat: 1549383904,
        id: "5c52379d9be6fc0017afd46e",
        name: "Sandwich Man"
      }
    }
  };
  wrapper = shallow(<PrivateRoute {..._.assign({}, props)} />);
  instance = wrapper.instance();
});

describe("render", () => {
  it("should render correctly", () => {
    expect(instance).toMatchSnapshot();
  });
});
