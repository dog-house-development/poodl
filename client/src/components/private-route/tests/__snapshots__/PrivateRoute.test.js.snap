// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`render should render correctly 1`] = `null`;
